ext/crates/fp/src/matrix/subquotient.rs

```diff
@@ -161,10 +166,12 @@ impl Subquotient {
     ///  * `subspace` - If this is None, it is empty
     pub fn from_parts(mut sub: Subspace, quotient: Subspace) -> Self {
         let dim = sub.dimension();
-        for row in sub.matrix.iter_mut().take(dim) {
+        let mut sub_matrix = sub.matrix_mut();
+
+        for row in sub_matrix.iter_mut().take(dim) {
             quotient.reduce(row.as_slice_mut());
         }
-        sub.matrix.row_reduce();
+        drop(sub_matrix);
```

Comment:
It seems a bit brittle to me to require explicitly dropping the guard to avoid panicking. Two other options I could see is making the `drop` implicit by putting the guard acquisition in its own block, or replacing the guard mechanism altogether by some function that would run a closure with a mutable matrix and row-reduce on exit. Now that I think about it, that second option might be more ergonomic

Reply:
> a bit brittle to me to require explicitly dropping the guard to avoid panicking

But the goal is to ensure two things:
1. the user cannot access a view into a subspace that isn't row reduced (because they hold a mutable borrow)
2. user code is reasonably explicit about the row reduction step
Ideally you'd want to make it a compile error for them to not explicitly call row reduce. But I don't know if that is possible?

Reply:
How about a closure-based accessor like in b112d77?

Reply:
Looks reasonable. But maybe we could call it something like `update_then_row_reduce`? I think that's pretty explicit about what is happening.

Reply:
Fair enough, I pushed that change